args-processor.go

```diff
@@ -0,0 +1,322 @@
+
+			signCh := make(chan os.Signal)
+			signal.Notify(signCh, os.Interrupt, syscall.SIGTERM)
+
+			select {
+			case err := <-errCh:
+				log.Fatal().Err(err).Msg("exitting")
+			case sign := <-signCh:
+				ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+				defer cancel()
+				_ = server.Shutdown(ctx)
+				ap.logger.Info().Str("signal", sign.String()).Msg("server stopped on signal")
+			}
+
+		}
+	}
+}
+
+func (ap *argsProcessor) generation() cli.CmdInitializer {
+	return func(cmd *cli.Cmd) {
+		modSrc := cmd.StringOpt("source", "", "Modules list source. Omit this option to use stdin")
+		gomodDir := cmd.StringOpt("gomod-dir", "", "Directory to save go.mod files in")
+		srcDir := cmd.StringOpt("source-dir", "", "Directory so save source archives in")
+		aprioriDest := cmd.StringOpt("dest", "", "File to save apriori info. Omit this option to use stdout")
+		recursive := cmd.BoolOpt("r recursive", false, "Download dependecies as well")
+		cmd.Spec = "[--recursive | -r] [--source=<source path>] [--dest=<apriori info file>] --gomod-dir=<go modules dir> --source-dir=<source archives dir>"
+		cmd.Action = func() {
+			if err := checkIfDir("invalid go modules directory (--gomod-dir)", *gomodDir); err != nil {
+				ap.logger.Fatal().Msgf(err.Error())
+			}
+			if err := checkIfDir("invalid source directory (--source-dir)", *srcDir); err != nil {
+				ap.logger.Fatal().Msgf(err.Error())
+			}
+
+			var src *os.File
+			if len(*modSrc) == 0 {
+				src = os.Stdin
+			} else {
+				file, err := os.Open(*modSrc)
+				if err != nil {
+					ap.logger.Fatal().Err(err).Msg("failed to open a source")
+				}
+				src = file
+			}
+
+			var dest *os.File
+			if len(*aprioriDest) == 0 {
+				dest = os.Stdout
+			} else {
+				file, err := os.Create(*aprioriDest)
+				if err != nil {
+					ap.logger.Fatal().Err(err).Msg("failed to open apriori destination")
+				}
+				dest = file
+			}
+
+			defer func() {
+				if err := src.Close(); err != nil {
+					ap.logger.Error().Err(err).Msg("failed to close source file")
+				}
+				if err := dest.Close(); err != nil {
+					ap.logger.Error().Err(err).Msg("failed to close dest file")
+				}
+			}()
+
+			mapping := apriori.Mapping{}
+			if err := ap.createAprioriInfo(modinfo.NewChannelFromSource(src.Name(), src), *gomodDir, *srcDir, *recursive, mapping); err != nil {
+				ap.logger.Fatal().Err(err).Msg("failed to generate apriori info")
+			}
+
+			data, err := json.MarshalIndent(mapping, "", "  ")
+			if err != nil {
+				ap.logger.Fatal().Err(err).Msg("failed to serialize apriori info")
+			}
+			if _, err := dest.Write(data); err != nil {
+				ap.logger.Fatal().Err(err).Msg("failed to save apriori info")
+			}
+			ap.logger.Info().Msg("apriori generation done")
+		}
+	}
+}
+
+func (ap *argsProcessor) getModule(module string) (goproxy.Module, error) {
+	req, _ := http.NewRequest("GET", fmt.Sprintf("http://society.org/%s/@v/list", module), nil)
```

Comment:
society.org? O_O

Reply:
It does nothing. We just need `http.Request` to get `goproxy.Module` from `gorpoxy.Plugin`